specifyweb/frontend/js_src/lib/components/QueryBuilder/Fields.tsx

```diff
@@ -155,11 +155,11 @@ export function QueryFields({
         fieldsContainerRef.current.clientHeight !==
           fieldsContainerRef.current.scrollHeight &&
         fields.length > oldFieldCount.current
-      )
-        scrollIntoView(
-          fieldsContainerRef.current.lastChild as HTMLElement,
-          'nearest'
-        );
+      ) {
+        const lastElement = fieldsContainerRef.current.lastChild as HTMLElement;
+        const firstNonContentsChild = lastElement.querySelector('button');
```

Comment:
could you try instead of `'button'` selector having a `':not(.contents)'` selector?
that would be more future-proof